lib/queuePopulator/LogReader.js

```diff
@@ -522,6 +484,94 @@ class LogReader {
         return undefined;
     }
 
+    _processLogEntry(batchState, record, entry, cb) {
+        function _executeFilter(ext, entry, cb) {
+            if (typeof ext.filterAsync === 'function') {
+                ext.filterAsync(entry, cb);
+            } else {
+                ext.filter(entry);
+                process.nextTick(cb);
+            }
+        }
+        async.eachSeries(this._extensions, (ext, next) => {
+            /**
+             * Delete entries don't have a value field.
+             * Value field normally contains the timestamp
+             * of the event, which is needed for the notification
+             * extension.
+             * That gets added here, we use the oplog event timestamp
+             */
+            if (
+                entry.type === 'delete' &&
+                ext.constructor.name === 'NotificationQueuePopulator'
+            ) {
+                entry.value = JSON.stringify({
+                    'last-modified': record.timestamp
+                });
+            // skipping entry for other extensions
+            } else if (entry.value === undefined) {
+                return next();
+            }
+            const entryToFilter = {
+                type: entry.type,
+                bucket: record.db,
+                // removing the v1 metadata prefix if it exists
+                key: transformKey(entry.key),
+                value: entry.value,
+                logReader: this,
+            };
+            return _executeFilter(ext, entryToFilter, next);
+        }, cb);
+    }
+
+    _filterEntries(batchState, record, cb) {
+        const { logStats } = batchState;
+
+        return async.eachSeries(record.entries, (entry, next) => {
+            logStats.nbLogEntriesRead += 1;
+            return this._processLogEntry(batchState, record, entry, next);
+        }, cb);
+    }
+
+    _processFilterEntry(batchState, record, done) {
+        const { entriesToPublish } = batchState;
+        if (!record || !record.entries) {
+            return done();
+        }
+        this._setEntryBatch(entriesToPublish);
+        return this._filterEntries(batchState, record, err => {
+            this._unsetEntryBatch();
+            if (err) {
+                this.log.error('error filtering entries from log', {
+                    method: 'LogReader._processFilterEntries',
+                    error: err,
+                });
+                return done(err);
+            }
+            return done();
+        });
+    }
+
+    _processFilterEntries(batchState, done) {
+        const { currentRecords } = batchState;
+        if (!currentRecords || currentRecords.length === 0) {
```

Comment:
Maybe safer like that, in case length is not defined?

```suggestion
        if (!currentRecords || !currentRecords.length) {
```